sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -888,6 +888,158 @@ case class MapFromEntries(child: Expression)
     copy(child = newChild)
 }
 
+case class MapSort(base: Expression, ascendingOrder: Expression)
```

Comment:
```suggestion
case class MapSort(base: Expression, ascendingOrder: Boolean)
```

Reply:
Doesn't the `BinaryExpression` require two expressions here? Do we demote this to UnaryExpression?

Reply:
What's is the internal use-cases for the expression? Do we need this parameter at all?

Seems like you are going to pass `true` as `ascendingOrder` always at
https://github.com/apache/spark/pull/45549/files#diff-11264d807efa58054cca2d220aae8fba644ee0f0f2a4722c46d52828394846efR2488

```scala
   case a @ Aggregate(groupingExpr, x, b) =>
      val newGrouping = groupingExpr.map { expr =>
        (expr, expr.dataType) match {
          case (_: MapSort, _) => expr
          case (_, _: MapType) =>
            MapSort(expr, Literal.TrueLiteral)
          case _ => expr
```

Reply:
From the point of internal use, we don't need it. Refactored expression as `UnaryExpression` and removed ordering altogether.